File: 2566-maximum-difference-by-remapping-a-digit/2566-maximum-difference-by-remapping-a-digit.java
class Solution {
    public int minMaxDifference(int num) {
        char ch[] =String.valueOf(num).toCharArray();
        int index=-1;
        
        for(int i=0; i<ch.length;i++)
        {
            if(ch[i]!='9')
            {
                index=i;
                break;
            }
        }
        if(index==-1)
        {
            return num;
        }
        int max=Integer.parseInt(String.valueOf(num).replaceAll(""+ch[index],"9"));
        int min=Integer.parseInt(String.valueOf(num).replaceAll(""+ch[0],"0"));
        return max-min;
    }
}

File: 2566-maximum-difference-by-remapping-a-digit/2566-maximum-difference-by-remapping-a-digit.cpp
class Solution {
public:
    int minMaxDifference(int num) {
     string num1=to_string(num);
        string num2=to_string(num);
        
        char c1=num1[0];
        char c2=num2[0];
        
        for(int i=0; i<num1.size(); i++)
        {
            if(num1[i]!='9')
            {
                c1=num1[i];
                break;
            }
        }
        for(int i=0; i<num1.size(); i++)
        {
            if(num1[i]==c1)
            {
                num1[i]='9';
            }
        }
        for(int i=0; i<num1.size();i++)
        {
            if(num2[i]==c2)
            {
                num2[i]='0';
            }
        }
        return stoi(num1)-stoi(num2); //max-min
    }
};